server/controllers/userController.js

```diff
@@ -334,6 +334,51 @@ class UserController {
       favoriteBooks: updatedUser.favoriteBooks
     })
   }
+
+  static async favouriteAnAuthor(req, res) {
+    const { id } = req.params;
+    const { loggedinUser } = req;
+
+    const requestedAuthor = await models.Authors.findByPk(id, {
+      attributes: ['firstName', 'lastName']
+    });
+
+    if (!requestedAuthor) {
+      return util.errorStatus(res, 400, 'The requested Author is available');
+    }
+
+    const { firstName, lastName } = requestedAuthor;
+
+    if (loggedinUser.favouriteAuthors) {
+      const alreadyfavourite = loggedinUser.favouriteAuthors.find(
+        author => author === Number(id)
+      );
+      if (alreadyfavourite) {
+        return util.successStatus(
+          res,
+          200,
```

Comment:
What do you think about status code 409 for this case?

Reply:
I didn't want the user to think of it as an error. Just like signing up with social media no errors if the user already exist then just log the user in.